### DISCUSSION TITLE
Incompatibility with MassPrunable trait

### DISCUSSION THREAD

**Discussion by Outsidaz** [jenssegers/laravel-mongodb#2597] [Q&A] [answered]
When I tried to use [model pruning](https://laravel.com/docs/10.x/eloquent#pruning-models) I got the logic exception:

`LogicException: Delete limit can be 1 or null (unlimited). Got 1000`

It looks like `Illuminate\Database\Eloquent\MassPrunable` is incompatible with `MongoDB\Laravel\Eloquent\Model`.

Any ideas? Could you write MassPrunable trait for the MongoDB model, please?

Thanks.

**Comment by GromNaN** [comment]
Hello @Outsidaz, thank you for the feedback. This exception was introduced last week by #2591.

MongoDB doesn't support [limit on delete operations](https://www.mongodb.com/docs/manual/reference/command/delete/#std-label-deletes-array-limit), so the chunk feature is irrelevant.
You can do this:

```php
(new Model)->prunable()->delete();
```

A new trait `MongoDB\Laravel\Eloquent\MassPrunable` could be created to support soft-delete without the chunk feature.

**Reply by GromNaN** [reply, marked as answer]
I opened a PR to implement MassPrunable with MongoDB compatibility. https://github.com/mongodb/laravel-mongodb/pull/2598
